experiment/autobumper/bumper/bumper.go

```diff
@@ -89,28 +264,126 @@ func UpdatePullRequestWithLabels(gc github.Client, org, repo, title, body, match
 	return nil
 }
 
-// UpdateReferences update the references of prow-images and testimages
-// in the files in any of "subfolders" of the current dir
+// UpdateReferences update the references of prow-images and/or testimages
+// in the files in any of "subfolders" of the includeConfigPaths but not in excludeConfigPaths
 // if the file is a yaml file (*.yaml) or extraFiles[file]=true
-func UpdateReferences(subfolders []string, extraFiles map[string]bool) (map[string]string, error) {
+func updateReferences(bumpProwImages, bumpTestImages bool, targetVersion string,
+	includeConfigPaths []string, excludeConfigPaths []string, extraFiles []string) (map[string]string, error) {
 	logrus.Info("Bumping image references...")
-	filter := regexp.MustCompile(prowPrefix + "|" + testImagePrefix)
-
-	for _, dir := range subfolders {
-		err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
-			if strings.HasSuffix(path, ".yaml") || extraFiles[path] {
-				if err := bumper.UpdateFile(path, filter); err != nil {
-					logrus.WithError(err).Errorf("Failed to update path %s.", path)
-				}
-			}
-			return nil
-		})
+	filters := make([]string, 0)
+	if bumpProwImages {
+		filters = append(filters, prowPrefix)
+	}
+	if bumpTestImages {
+		filters = append(filters, testImagePrefix)
+	}
+	filterRegexp := regexp.MustCompile(strings.Join(filters, "|"))
+
+	var tagPicker func(string, string, string) (string, error)
+	var err error
+	switch targetVersion {
+	case latestVersion:
+		tagPicker = imagebumper.FindLatestTag
+	case upstreamVersion:
+		tagPicker, err = upstreamImageVersionResolver(upstreamBranch + "/" + prowRefConfigFile)
+		if err != nil {
+			return nil, err
+		}
+	case upstreamStagingVersion:
+		tagPicker, err = upstreamImageVersionResolver(upstreamBranch + "/" + prowStagingRefConfigFile)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		tagPicker = func(imageHost, imageName, currentTag string) (string, error) { return tagVersion, nil }
+	}
+
+	updateFile := func(name string) error {
+		fmt.Printf("Updating file %s\n", name)
+		if err := imagebumper.UpdateFile(tagPicker, name, filterRegexp); err != nil {
+			logrus.WithError(err).Errorf("Failed to update file %q.", name)
```

Comment:
Logging is unnecessary here. The caller can decide whether or not to log the error. We also do not need to pass in `name` because the caller knows this information already.

Reply:
Removed this logging line.